test/orm/test_mapper.py

```diff
@@ -1996,6 +1997,38 @@ class B(OldStyle, NewStyle):
 
         mapper(B, users)
 
+    @testing.provide_metadata
+    def test_readonly_setter(self):
+        t = Table(
+            "foo-table",
+            self.metadata,
+            Column("id", Integer(), primary_key=True),
+            Column("comp", Integer(), sa.Computed("id + 1")),
+        )
+
+        class A(object):
+            pass
+
+        mapper(A, t, properties={"prop": column_property(t.c.id - 1)})
+
+        a = A()
+        a.id = 3
+
+        def setComputed():
+            a.comp = 33
+
+        def setProp():
+            a.comp = 33
+
+        assert_warnings(
```

Comment:
I prefer we use the context manager version here
```

with testing.expect_warnings("the value set to readonly..."):
    a.comp = 33


```

Reply:
definitely better